### DISCUSSION TITLE
Voucher plugin - Why does the amount option not loop through the cart like the percentage option?

### DISCUSSION THREAD

**Discussion by Rodien** [openstore-ecommerce/OpenStore#263] [Q&A]
Hi guys,

Is there any reason why the amount option does not loop through all the items? 

Let's say productA costs 20EU but I have a discountcode of 10eu.

If I add productA twice in my cart I expect to get a total discount amount of 20EU.

![image](https://user-images.githubusercontent.com/50739829/161908447-f39cfe11-debf-4649-9659-5fa0192b81fc.png)

I must add that I am currently working on voucher plugin with more options than the default openstore one.
With this plugin you can apply a discount to products under a specfic category.

![image](https://user-images.githubusercontent.com/50739829/161910156-1c81809f-97ad-49ef-8c71-a32ace8abe80.png)

**Comment by Rodien** [comment]
Is there a way I can override this process?

We can have something like apply this discount on the whole cart or per items who qualify

Take a look at Nevoweb.DNN.NBrightBuy.Components.CartData ValidateCart() method

**Comment by DNNMonster** [comment]
Possibly with custom templates and/or custom tokens but I'm not sure...  It may be not be a very elegant approach.  The discountcodeamt calculated data point may require changes to NBrightBuy to achieve the "apply discount per model/item".   I think if the setting were added we could modify the calc function here to achieve the goal: https://github.com/openstore-ecommerce/OpenStore/blob/ce09324695cc817ceb49919fa0dfc902950595ea/Providers/DiscountCodesProvider/DiscountCodesProvider.cs#L87

It looks like a relatively low risk change but I've been wrong before about that.

**Reply by Rodien** [reply]
Thanks for the reply.

Yeah I hope we can find a solution that won't break something else.
